### PR TITLE
[mds-agency] [mds-agency-cache] 1.0.0 Agency Tests Passing

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
Gets mds-agency successfully passing preexisting tests. This does *not* add additional tests to cover new use-cases for 1.0 yet. 

Please take a careful look at this, especially the changes to the test suite itself. There are a few places where I had to do some pretty janky stuff (mostly around extracting request body for writing vehicle events), if you have clean suggestions please post them :)

## 📦 Impacts:
[mds-agency]
[mds-agency-cache]

### CODE REVIEW

File: packages/mds-agency/tests/integration-tests.spec.ts [hunk 1]
```diff
@@ -705,7 +710,8 @@ describe('Tests API', () => {
       .post(pathPrefix(`/vehicles/${TRIP_UUID}/event`))
       .set('Authorization', AUTH)
       .send({
-        event_type: VEHICLE_EVENTS.maintenance_pick_up,
+        event_types: [VEHICLE_EVENTS.maintenance_pick_up],
```
Comment (drtyh2o) [inline packages/mds-agency/tests/integration-tests.spec.ts:713]: Can we change `VEHICLE_EVENTS` and other enums to use `const` assertion like all of the newer type do:
```ts
const VEHICLE_EVENTS = ['maintenance_pick_up', ...] as const
type VEHICLE_EVENT = typeof VEHICLE_EVENTS[number]
```

Comment (avatarneil) [reply packages/mds-agency/tests/integration-tests.spec.ts:713]: Yep, after merging this that's next up on my list of things to do to prep for Taxi.

File: packages/mds-agency/tests/integration-tests.spec.ts [hunk 2]
```diff
@@ -1274,8 +1277,8 @@ describe('Tests API', () => {
         test.value(deviceA.device_id).is(DEVICE_UUID)
         test.value(deviceA.provider_id).is(TEST1_PROVIDER_ID)
         test.value(deviceA.gps.lat).is(TEST_TELEMETRY.gps.lat)
-        test.value(deviceA.status).is(VEHICLE_STATES.available)
-        test.value(deviceA.prev_event).is(VEHICLE_EVENTS.reservation_cancel)
+        test.value(deviceA.state).is(VEHICLE_STATES.available)
+        test.value(JSON.stringify(deviceA.prev_events)).is(JSON.stringify([VEHICLE_EVENTS.reservation_cancel]))
```
Comment (janedotx) [inline packages/mds-agency/tests/integration-tests.spec.ts:1281]: Nit, but I don't see why you have to stringify both things.

Comment (avatarneil) [reply packages/mds-agency/tests/integration-tests.spec.ts:1281]: Array equality in javascript is tricky https://masteringjs.io/tutorials/fundamentals/compare-arrays

Comment (avatarneil) [reply packages/mds-agency/tests/integration-tests.spec.ts:1281]: Using JSON.stringify is a bit of a shortcut and has some edge cases that cause problems, but for the sake of this test we don't expect `null | undefined` values in the arrays at all, so should be fine.

File: packages/mds-agency/tests/integration-tests.spec.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-agency/tests/integration-tests.spec.ts]: Merged in a PR which makes this switch
